pkg/yttlibrary/overlay/array_item_match_annotation.go

```diff
@@ -76,6 +83,10 @@ func (a ArrayItemMatchAnnotation) Indexes(leftArray *yamlmeta.Array) ([]int, err
 }
 
 func (a ArrayItemMatchAnnotation) MatchNodes(leftArray *yamlmeta.Array) ([]int, []*filepos.Position, error) {
+	if a.unannotated {
+		return []int{}, []*filepos.Position{leftArray.GetPosition()}, nil
```

Comment:
this should be `return nil, nil, nil` since there are no matches.

Reply:
You're right, there are no matches... The intent of this change is to allow the `MatchAnnotationExpectsKwarg.Check` function to pass without error. This may be the wrong way, or it may have been too indirect.

We are really trying to return no matches, without erroring, which would mean we don't have to change 
`mergeArrayItem` in `array.go`:

```Golang
leftIdxs, err := ann.Indexes(leftArray)
if err != nil {
	if err, ok := err.(MatchAnnotationNumMatchError); ok && err.isConditional() {
		return nil
	}
	return err
}

if len(leftIdxs) == 0 {
	return o.appendArrayItem(leftArray, newItem)
}
```

Reply:
are you saying that `a.expects.Check(matches)` fails because it expects 1 match? i imagine then we should  configure a.expects to expect 0 matches so that its Check succeeds without error.

Reply:
That's a good idea. I'm envisioning a change in the place that sets `unannotated = true`, where instead of needing that extra value, we would create a different `expects` for the matchAnnotation. That different "expects" could expect no matches, and return a nil error, which would then nicely default in array.go